### DISCUSSION TITLE
Decontx: Heuristic clustering or z parameter? How are the cluster labels specified with the z parameter affecting the decontamination?

### DISCUSSION THREAD

**Discussion by MartaSanchezCarbonell** [campbio/celda#379] [Q&A] [answered]
Dear Cambell Lab,

I am using your tool DecontX to decontaminate my droplet scRNA-seq dataset and I came up with some doubts.
In https://bioconductor.org/packages/3.16/bioc/vignettes/celda/inst/doc/decontX.html#running-decontx , when decontx is applied, this statement is said: 

"decontX will perform heuristic clustering to quickly define major cell clusters. However if you have your own cell cluster labels, they can be specified with the z parameter."

I would like to ask first, which strategy would you recommend better? Using the heuristic clustering, I obtain a much higher contamination fraction than specifying them with the z parameter. But, I am unsure about using the z parameter because, then, the contamination fraction depends on the number of clusters that I supply to the function (the contamination fraction decreases when the number of clusters increases ). How can I determine the best strategy and, if using the z parameter, the best number of clusters?

Thank you very much in advance and thank you for this great tool!

Best regards,

Marta

**Comment by joshua-d-campbell** [comment, marked as answer]
Hi @MartaSanchezCarbonell, thanks for trying out our tool! Your question is a good one. DecontX assumes that each cell is a mixture of counts from its true cell population and from ambient RNA. It uses the cluster labels to estimate what the true cell population should look like. Our quick/heuristic clustering procedure will tend to identify "broader" cell populations. For example all T-cells may be grouped into a single cluster whereas if you did a different clustering workflow like Seurat, the T-cells may get split up into multiple clusters (e.g. CD8, CD4, etc). 

There are also two ways in which we can estimate the distribution of ambient RNA distribution. The first is when the raw/background matrix is directly supplied via the `background` parameter. In the second approach, when the raw/background matrix is **not** supplied, we can estimate the what the ambient RNA distribution most likely looks like from the other cell clusters. For example, if you have several cell clusters other than T-cells (e.g. epithelial, myeloid, etc), the ambient RNA profile for the T-cells will be estimated as a linear combination of these other populations. 

Now getting back to your question. If you supply more granular cluster labels via the `z` parameter, this can *sometimes* have the effect of estimating less contamination within a cluster. This is because other similar clusters are now included in the cells used to estimate the ambient RNA profile for that cluster. For example, if you have a clustering solution that splits up CD8 and CD4 T-cells, then the CD4 T-cells will be included in the estimation of the ambient RNA profile for CD8 T-cells (and visa versa). This can have the effect of "down-weighting" genes for the non-native cell types (e.g. markers for epithelial, myeloid, etc.) in the ambient profile for that CD8 T-cell cluster.

Note that using the raw/background matrix may also have a similar effect for some of the more predominant cell populations as they are the populations contributing the most RNA to the ambient profiles in the empty drops. 

The other challenging scenario is when a dataset has a transitionary or intermediate cell population that is "in-between" two other cell populations. This cluster can be flagged as having high contamination when it is clustered with other cell populations using DecontX's default clustering method. It may be better to make this group of cells have its own cluster label rather than lumping it in with the other cells. But this is situational and may depend on how likely you believe this is a real cluster and not just a group of highly contaminated cells. 

I hope that helps explain the algorithm a bit more. It may be hard to give you more guidance without seeing the specifics of you dataset. As this question may come up a lot, I am going to move this thread to a Discussion instead of an Issue.

**Reply by MartaSanchezCarbonell** [reply]
Dear @joshua-d-campbell , thank you very much for your answer! It is true that I haven't given any information about my kind of dataset, I am analysing brain samples from different timepoints focused on development (10.000 cells each timepoint). Probably I will have lots of cells in transient states and my research is also focused on that, on studying the nature of cell heterogeneity (if this heterogeneity is given by having the same cells in different functional states or if we can differentiate them in "real" different cell types). 

We are using the row count matrix (without the Cell Ranger filtering) as the background parameter and the count matrix being filtered by cell ranger as X argument.  After decontamination, we proceed with an adapted pre-processing pipeline based on Seurat, with a first step to filter out all the bad quality cells, then normalisation, etc, until clustering analysis. So there are 2 filtering steps where I remove barcodes that I don't consider real meaningful cells (at Cell Ranger and at Seurat pre-processing). I explain this because, by reading your answer and informing myself on internet, I have another doubt also related with the cluster topic. When I apply DecontX, I do it on the count matrix after the Cell Ranger filtering (so I still have some barcodes that will be removed with my Seurat filtering, normally I discard 3000 barcodes in this filtering step), and for this matrix I normally don't have clustering information. Furthermore, even if I take this matrix and I cluster it (without the Seurat filtering and with still contamination) I am not sure how reliable would be that clustering. I would like to ask for advice because the majority of examples that I've seen are giving the clustering information in a really precise way and I am not sure how. I could simply do a clustering but I see it really difficult, having all the high and low quality cells playing a part in the algorithm. Could you please comment on that?

Thank you very much again for your help and I am eager to learn from your knowledge and experience

**Reply by joshua-d-campbell** [reply]
Hi @MartaSanchezCarbonell, this can be tricky. We have also had some situations where we perform some additional quick filtering on the filtered counts matrix (e.g. filter cells out with low UMI and low number of features detected) and the subsequently run decontX. This was because the Cellrangner was probably not stringent enough on calling what is a good quality cell. You may have to try several combinations of procedures to see what produces the cleanest clusters. If you do additional filtering before running decontX, I don't see that as being a huge issue. You may also have to do the clustering twice, once with a larger number of cells and then subsequently after filtering out more cells.
